examples/platforms/da15000/radio.c

```diff
@@ -0,0 +1,551 @@
+
+exit:
+    return error;
+}
+
+RadioPacket *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
+{
+    (void)aInstance;
+    return &s_transmit_frame;
+}
+
+ThreadError otPlatRadioTransmit(otInstance *aInstance, RadioPacket *aPacket)
+{
+    ThreadError error = kThreadError_None;
+    (void)aInstance;
+
+    FTDF_Boolean csmaSuppress = 0;
+    VerifyOrExit(s_state == kStateReceive, error = kThreadError_Busy);
+
+    ad_ftdf_send_frame_simple(aPacket->mLength, aPacket->mPsdu, aPacket->mChannel, 0, csmaSuppress); //Prio 0 for all.
+    s_state = kStateTransmit;
+    s_data_pending = false;
+#if DEBUG_LOG_ENABLE
+    hw_uart_write_buffer(HW_UART1, "\nT", 2);
+#endif
+
+exit:
+    return error;
+}
+
+
+int8_t otPlatRadioGetRssi(otInstance *aInstance)
+{
+    (void)aInstance;
+    return 0;
+}
+
+otRadioCaps otPlatRadioGetCaps(otInstance *aInstance)
+{
+    (void)aInstance;
+    return kRadioCapsNone;
+}
+
+void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
+{
+    (void)aInstance;
+    uint8_t value;
+    value = aEnable;
+    FTDF_setValue(FTDF_PIB_PROMISCUOUS_MODE, &value);
+    s_promiscuous = aEnable;
+}
+
+bool otPlatRadioGetPromiscuous(otInstance *aInstance)
+{
+    (void)aInstance;
+    return s_promiscuous;
+}
+
+ThreadError otPlatRadioHandleTransmitDone(bool *rxPending)
+{
+    ThreadError error = kThreadError_None;
+    VerifyOrExit(s_state == kStateTransmit, error = kThreadError_InvalidState);
+
+    s_state = kStateReceive;
+
+    if (rxPending != NULL)
+    {
+        *rxPending = s_data_pending;
+    }
+
+exit:
+    return error;
+}
+
+
+
+
+void FTDF_sendFrameTransparentConfirm(void         *handle,
+                                      FTDF_Bitmap32 status)
+{
+    (void)handle;
+    (void)status;
+    volatile uint32_t *txStatus = FTDF_GET_REG_ADDR_INDEXED(RETENTION_RAM_TX_RETURN_STATUS_1, FTDF_TX_DATA_BUFFER);
+
+    SendFrameDone = true;
+
+#if DEBUG_LOG_ENABLE
+    hw_uart_write_buffer(HW_UART1, "\nTD", 3);
+#endif
+}
+
+
+
+void da15000RadioProcess(otInstance *aInstance)
+{
+    bool rxPending;
+    uint8_t value = 0;
+
+    if (SendFrameDone)
+    {
+        FTDF_SET_FIELD(ON_OFF_REGMAP_RXENABLE, 1);
+        otPlatRadioHandleTransmitDone(&rxPending);
+        otPlatRadioTransmitDone(aInstance, &s_transmit_frame, false, kThreadError_None);
+        SendFrameDone = false;
+    }
+
+    if (SED)
+    {
+        if ((otPlatAlarmGetNow() - sleep_const_delay) > 100)
+        {
+            FTDF_setValue(FTDF_PIB_RX_ON_WHEN_IDLE, &value);
+            s_state = kStateSleep;
+            SED = false;
+        }
+
+    }
+}
+
+
+
+void FTDF_rcvFrameTransparent(FTDF_DataLength frameLength,
+                              FTDF_Octet     *frame,
+                              FTDF_Bitmap32   status)
+{
+    (void)status;
+    RadioPacket otRadioPacket;
+    //  int8_t        value;
+    //  value  = -20;
+
+    otRadioPacket.mPower = -20;           //FIX THIS -actual measurements shall be placed here
```

Comment:
This variable should probably be populated with real data before it is added to master.